pkg/csi/service/cns/nodes.go

```diff
@@ -58,7 +58,20 @@ func (nodes *Nodes) nodeAdd(obj interface{}) {
 }
 
 func (nodes *Nodes) nodeUpdate(oldObj interface{}, newObj interface{}) {
-	nodes.nodeRegister(newObj)
+	newNode, ok := newObj.(*v1.Node)
+	if !ok {
+		klog.Warningf("nodeUpdate: unrecognized object newObj %[1]T%+[1]v", newObj)
+		return
+	}
+	oldNode, ok := oldObj.(*v1.Node)
+	if !ok {
+		klog.Warningf("nodeUpdate: unrecognized object newObj %[1]T%+[1]v", oldObj)
```

Comment:
Hi @chethanv28,

I updated my comment because the original had the wrong string (I copy-pasted). Looks like you copied mine, which had the problem. This log should be `unrecognized object oldObj` I believe.

Reply:
Correct, Thanks